blockchain/chain/src/store/chain_store.rs

```diff
@@ -17,7 +18,7 @@ pub struct ChainStore<'a> {
     // TODO add a pubsub channel that publishes an event every time the head changes.
 
     // key-value datastore
-    db: Blockstore,
+    pub db: Blockstore,
```

Comment:
you have the function `blockstore()` already, so why do you need to make this public?

Reply:
I need access to the underlying DB type, for 

` let bls_root = AMT::new_from_slice(&self.chain_store.db, &bls_cids)?;`

If I made use of `blockstore()`, such as 

`let bls_root = AMT::new_from_slice(&self.chain_store.blockstore().db()?, &bls_cids)?;`

`AMT::new_from_slice` complains about the reference and error type.

Reply:
yeah, because it was behind two references (I don't know where you got that function) use:
```
        let bls_root = AMT::new_from_slice(self.chain_store.blockstore(), &bls_cids)?;
        let secp_root = AMT::new_from_slice(self.chain_store.blockstore(), &secp_cids)?;
```

and you don't need to make the db variable public in two different ways